Fraction_class.py

```diff
@@ -0,0 +1,87 @@
+#def a Fraction class, which has two attributes: numerator and denomitor, and three methods: __str__,__add__,__eq__
+#understand the differences between shallow equation and deep equation
+def gcd_euclid_recursion(m,n):
+    m=abs(m)
+    n=abs(n)
+    if m==n:

```

Comment:
logically this is not necessary, as if `m == n`, m % n == 0, covered by logic below.

Reply:
make sense!